include/gatekeeper_varip.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GATEKEEPER_VARIP_H_
+#define _GATEKEEPER_VARIP_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/*
+ * The code of this file is mostly a copy of the Linux kernel.
+ */
+
+/* TODO Add support for variable IP headers for all the functional blocks. */
```

Comment:
Yes, I have added this functionality for ``LLS``, ``CPS``, and ``GGU`` blocks. Maybe the comment is misleading. Here, I just take a note that one needs to implement this functionality in case some functional blocks need to support more protocols. I just changed the comment.